wdl/PBCLRDemultiplexOnlySingleFlowcell.wdl

```diff
@@ -40,48 +42,57 @@ workflow PBCLRDemultiplexOnlySingleFlowcell {
         String DIR = SM + "." + ID
 
-        # demultiplex CLR BAM
-        call PB.Demultiplex { input: bam = subread_bam, ccs = false, guess = 0, prefix = "~{SM}.~{ID}", barcode_file = barcode_file }
+        # shard one raw BAM into fixed chunk size (num_reads_per_split)
+        call Utils.ShardLongReads { input: unmapped_files = [ subread_bam ], num_reads_per_split = num_reads_per_split }
 
-        # make reports on demultiplexing
-        call PB.MakeSummarizedDemultiplexingReport as SummarizedDemuxReportPNG { input: report = Demultiplex.report }
-        call PB.MakeDetailedDemultiplexingReport as DetailedDemuxReportPNG { input: report = Demultiplex.report, type="png" }
-        call PB.MakeDetailedDemultiplexingReport as DetailedDemuxReportPDF { input: report = Demultiplex.report, type="pdf" }
-        call PB.MakePerBarcodeDemultiplexingReports as PerBarcodeDetailedDemuxReportPNG { input: report = Demultiplex.report, type="png" }
-        call PB.MakePerBarcodeDemultiplexingReports as PerBarcodeDetailedDemuxReportPDF { input: report = Demultiplex.report, type="pdf" }
-
-        ##########
-        # store the results into designated bucket
-        ##########
-
-        call FF.FinalizeToDir as FinalizeDemuxReads {
-            input:
-                files = Demultiplex.demux_bams,
-                outdir = outdir + "/" + DIR + "/reads"
+        # then perform correction and alignment on each of the shard
+        scatter (subreads in ShardLongReads.unmapped_shards) {
+            # demultiplex CLR BAM shard
+            call PB.Demultiplex { input: bam = subreads, ccs = false, guess = 0, prefix = "~{SM}.~{ID}", barcode_file = barcode_file }
         }
 
-        call FF.FinalizeToDir as FinalizeLimaMetrics {
-            input:
-                files = [ Demultiplex.counts, Demultiplex.report, Demultiplex.summary ],
-                outdir = outdir + "/" + DIR + "/metrics/lima"
-        }
-
-        call FF.FinalizeToDir as FinalizeLimaSummary {
-            input:
-                files = SummarizedDemuxReportPNG.report_files,
-                outdir = outdir + "/" + DIR + "/figures/lima/summary/png"
-        }
-
-        call FF.FinalizeToDir as FinalizeLimaDetailedPNG {
-            input:
-                files = DetailedDemuxReportPNG.report_files,
-                outdir = outdir + "/" + DIR + "/figures/lima/detailed/png"
-        }
-
-        call FF.FinalizeToDir as FinalizeLimaDetailedPDF {
-            input:
-                files = DetailedDemuxReportPDF.report_files,
-                outdir = outdir + "/" + DIR + "/figures/lima/detailed/pdf"
-        }
+#        # merge the per-shard demultiplexed BAMs into one, corresponding to one raw input BAM
+#        call Utils.MergeBarcodeBams as MergeChunks { input: bams = Demultiplex.demux_bams }
+#
+#        # make reports on demultiplexing
+#        call PB.MakeSummarizedDemultiplexingReport as SummarizedDemuxReportPNG { input: report = Demultiplex.report }
```

Comment:
Since we're scattering over the `ShardLongReads.unmapped_shards`, `Demultiplex.report` will be an array.  Will these calls handle an array of report objects?

Reply:
Oh oops, this file wasn't supposed to be committed at all.  The multiplexing workflows are part of a separate branch (or at least are supposed to be).  I think you're right in that the report array is probably not being handled correctly yet.  I'll remove this file from this commit and resubmit.